### PR TITLE
fix(gatsby-telemetry): Ensure quickly running commands exit freely

### PR DESCRIPTION
**Pull request by jamo**
It turns out the events timer caused `gatsby clean` in particular to never exit - unlike other fairly quickly running commands such as help, telemetry and build...

This should fix it

### CODE REVIEW

File: packages/gatsby-cli/src/create-cli.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ const {
   setDefaultTags,
   setTelemetryEnabled,
 } = require(`gatsby-telemetry`)
+console.log(require.resolve(`gatsby-telemetry`))
```
Comment (pieh) [inline packages/gatsby-cli/src/create-cli.js]:
```suggestion
```

File: packages/gatsby/src/commands/develop.js [hunk 2]
```diff
@@ -258,6 +258,7 @@ async function startServer(program) {
 module.exports = async (program: any) => {
   initTracer(program.openTracingConfigFile)
   telemetry.trackCli(`DEVELOP_START`)
+  telemetry.startBackgroudUptate()
```
Comment (pieh) [inline packages/gatsby/src/commands/develop.js]: is this also needed (or useful) in `gatsby serve` which is second non-self-exiting gatsby command?

File: packages/gatsby-cli/src/create-cli.js (hunk 1 above)
Comment (jamo) [reply packages/gatsby-cli/src/create-cli.js]: So this ended up sneaking in after all, thanks for spotting it

File: packages/gatsby/src/commands/develop.js (hunk 2 above)
Comment (jamo) [reply packages/gatsby/src/commands/develop.js]: Indeed, we should add it for that too. 
My Mac is just installing updates so I'll update this tomorrow morning. Or if we want to ship this before it, feel free to edit the pr :)

Comment (sidharthachatterjee) [reply packages/gatsby/src/commands/develop.js]: Done, shipping now

**Pull request merged**